Tools/CMake/AMReXParallelBackends.cmake

```diff
@@ -211,6 +211,7 @@ if (AMReX_HIP)
    find_package(rocprim REQUIRED CONFIG)
    find_package(hiprand REQUIRED CONFIG)
    target_link_libraries(amrex PUBLIC hip::hiprand roc::rocrand roc::rocprim)
+   target_link_libraries(amrex PUBLIC $<$<LINK_LANGUAGE:CXX>:${HIP_LIBRARIES}>)
```

Comment:
Ok, the genex does not work here, but generally the content of `HIP_LIBRARIES` includes `hip::device` which is blocked by https://github.com/ROCm-Developer-Tools/HIP/pull/2190 as the PR in #2031